### DISCUSSION TITLE
Changing the build port number

### DISCUSSION THREAD

**Discussion by lamettrist** [nuxt/nuxt#24435] [Questions] [answered]
Hi there!

I am trying to build an desktop application utilizing Nuxt & Tauri, for which I want to have a port number other than 3000 for the application when compiled. How can I achieve this by changing the port number of the built application in nuxt.config?

**Comment by darioferderber** [comment]
Hi @De-y !

You can create an `.env` file in the root of the project with the PORT variable.
`.env` file can be different for development environment and production environment.

```
PORT=4000
```

**Reply by lamettrist** [reply]
Hi Dario, 

Inside of development, the port works correctly both using .env and by changing the port in config; however, when I build the code and then run it, the web server defaults to the production environment.

**Reply by AloisSeckar** [reply, marked as answer]
Thats because `.env` is [not being loaded in production mode](https://nuxt.com/docs/guide/directory-structure/env#production-preview)

The way of providing configuration to production site depends on the target platform. For example for Netlify deployment, there is a dedicated "Environment variables" section in site administration. I guess most cloud providers will have theirs alike.

**Reply by darioferderber** [reply]
I agree with @AloisSeckar, it depends on the platform.

If you're starting your application with `node .output/server/index.mjs` you can add `PORT=4000` before the command, like this `PORT=4000 node .output/server/index.mjs`

**Comment by xjccc** [comment]
try `NITRO_PORT=4000`

**Comment by masyoudi** [comment]
i have solution, maybe this is ugly , but its works properly

i have project structure like this
```
.
├── .output                              -- The build result
├── client                               -- The srcDir
├── config                               -- Config file
│     ├── .env
│     └── script.mjs                     -- Script to modified package.json in .output folder
├── server                               -- The server dir
├── nuxt.config.ts
└── package.json 
```


`/config/.env` file
```bash
PORT                 = 4033
HOST                 = "localhost"
```

`/config/script.mjs` file
```js
import { existsSync, mkdirSync, readFileSync, writeFileSync } from 'fs';
import { fileURLToPath } from 'url';
import { config as loadConfig } from 'dotenv';

const updateScript = (file) => {
  const data = JSON.parse(readFileSync(file, 'utf8'));
  const env = fileURLToPath(new URL('.env', import.meta.url));
  const { parsed } = loadConfig({ path: env, override: true });

  typeof parsed === 'undefined' ? console.warn('Failed parse .env in ', env) : void 0;

  const port = parsed?.PORT || 4033;
  const host = parsed?.HOST || '0.0.0.0';
  const start = `NODE_ENV=production PORT=${port} HOST=${host} node index.mjs`;
  const configPath = fileURLToPath(new URL('../.output/server/config', import.meta.url));

  if (!existsSync(configPath)) {
    mkdirSync(configPath);
  }

  writeFileSync(configPath.concat('/.env'), readFileSync(env, 'utf8'));

  data.scripts = { start: start };
  writeFileSync(file, JSON.stringify(data, null, 2));
};

(async () => {
  try {
    const pkg = fileURLToPath(new URL('../.output/server/package.json', import.meta.url));
    if (existsSync(pkg)) {
      updateScript(pkg);
      return;
    }

    console.error('Failed to find file: ', pkg);
  } catch {
    console.error('Failed to export script runner');
  }
})();
```

`package.json` file
```json
{
  "name": "app",
  "version": "1.0.0",
  "type": "module",
  "scripts": {
    "dev": "NODE_ENV=development nuxt dev --dotenv ./config/.env",
    "build": "NODE_ENV=production nuxt build",
    "start": "node config/script.mjs && cd .output/server && npm start"
  },
  "dependencies": {
    "dotenv": "^16.3.1",
    "nuxt": "^3.7.4"
  }
}
```

and just run `npm run build` & `npm start`

the point is to modified script of `package.json` inside `.output` folder
